### PR TITLE
Update CHANGELOG for a recent source-breaking fix to 'findProtocolSelfReferences'

### PR DESCRIPTION
**Pull request by AnthonyLatsis**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,29 @@ _**Note:** This is in reverse chronological order, so newer entries are added to
 Swift 5.6
 ---------
 
+* References to `Self` or so-called "`Self` requirements" in the type signatures
+  of protocol members are now correctly detected in the parent of a nested type.
+  As a result, protocol members that fall under this overseen case are no longer
```
Comment (benrimmington) [inline CHANGELOG.md:11]: `overseen` → `overlooked` ?

```suggestion
  As a result, protocol members that fall under this overlooked case are no longer
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,29 @@ _**Note:** This is in reverse chronological order, so newer entries are added to
...
+  extension P {
+    func method(arg: Outer<Self>.Inner) {}
+  }
+
+  func test(p: P) {
+    // error: 'method' has a 'Self' requirement and cannot be used on a value of
+    // protocol type (use a generic constraint instead).
+    p.method
+  }
+}
```
Comment (benrimmington) [inline CHANGELOG.md:29]: The extra closing brace is somehow causing the previous SE-0315 entry to be rendered as a code block.

```suggestion
```

**Review by benrimmington — COMMENTED** [review]
This PR may conflict with #39641.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,28 @@ _**Note:** This is in reverse chronological order, so newer entries are added to
...
+
+  protocol P {}
+  extension P {
+    func method(arg: Outer<Self>.Inner) {}
+  }
+
+  func test(p: P) {
+    // error: 'method' has a 'Self' requirement and cannot be used on a value of
+    // protocol type (use a generic constraint instead).
+    p.method
```
Comment (slavapestov) [inline CHANGELOG.md:27]: Perhaps `_ = p.method`?

**Review by slavapestov — APPROVED** [review]

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

**Pull request merged**